vassal-app/src/main/java/VASSAL/build/module/PrivateMap.java

```diff
@@ -209,6 +210,14 @@ else if (isVisibleTo(newSide)) {
       ((View)getView()).enableMotionListeners();
     }
     getLaunchButton().setEnabled(isVisibleTo(PlayerRoster.getMySide()));
+
+    // Close this private window if we've switched to a side not allowed to see it
+    final Container tla = theMap.getTopLevelAncestor();
+    if (tla != null) {
+      if (tla.isVisible() && !isVisibleTo(newSide)) {
```

Comment:
Is there a reason not to combine these two `if` conditions?

Reply:
fixed